src/Appwrite/Platform/Workers/Migrations.php

```diff
@@ -407,4 +408,28 @@ function () use ($migration, $transfer, $projectDocument, $queueForRealtime) {
             }
         }
     }
+
+    /**
+     * Stores a human-readable error message for migration.
+     *
+     * Only applies to migrations where the source is CSV.
+     */
+    private function setMigrationError(Document $migration, ?Source $source, ?Destination $destination): void
+    {
+        if ($migration->isEmpty() || $source === null || $destination === null) {
+            return;
+        }
+
+        $sourceErrors = $source->getErrors();
+        $destinationErrors = $destination->getErrors();
+        $migrationSource = $migration->getAttribute('source');
+
+        if ($migrationSource === CSV::getName()) {
+            if (! empty($sourceErrors)) {
+                $migration->setAttribute('error', $sourceErrors[0]->getMessage());
```

Comment:
Can we do this without a new attribute if we just get the first error's message?